pkg/admission/permissionclaims/mutating_permission_claims.go

```diff
@@ -129,7 +129,7 @@ func (m *mutatingPermissionClaims) Validate(ctx context.Context, a admission.Att
 		return err
 	}
 
-	expectedLabels, err := m.permissionClaimLabeler.LabelsFor(ctx, clusterName, a.GetResource().GroupResource(), a.GetName())
+	expectedLabels, err := m.permissionClaimLabeler.LabelsFor(ctx, clusterName, a.GetResource().GroupResource(), a.GetName(), a.GetNamespace())
```

Comment:
@s-urbaniak I discussed the namespace field changes with @ncdc and he recommended putting the accept/reject decision in this Validate method.

I'd like to surface the selection decision that happens in `LabelsFor` without re-fetching the APIBinding and iterating over the permission claims again.

Can we store those values in a struct field? Making `LabelsFor` return a third value seems like the wrong solution to me.

Reply:
can you elaborate on the re-fetching case? Is it because they have been already fetched in `Admit`? The only thing that comes to my mind is passing it via the context 🤔